### PR TITLE
RATIS-1333. Add explicit thread names for FollowerState and LeaderState threads.

### PR DESCRIPTION
**Pull request by jojochuang**
## What changes were proposed in this pull request?

A simple improvement to make make thread names more distinct, making troubleshooting easier. 

## What is the link to the Apache JIRA
https://issues.apache.org/jira/browse/RATIS-1333

## How was this patch tested?
Confirmed with a VisualVM output that shows thread names.

### CODE REVIEW

**Comment by jojochuang** [comment]
image

File: ratis-server/src/main/java/org/apache/ratis/server/impl/FollowerState.java [hunk 1]
```diff
@@ -153,4 +154,6 @@ public  void run() {
   public String toString() {
     return name;
   }
+
+
```
Comment (szetszwo) [inline ratis-server/src/main/java/org/apache/ratis/server/impl/FollowerState.java:158]:
```suggestion
```

**Review by szetszwo — APPROVED** [review]
+1 the change looks good.

There are some unrelated empty lines added.  Will revert it before merging.

**Pull request merged**